### DISCUSSION TITLE
No module named 'codechecker_api_shared' when CodeChecker Server

### DISCUSSION THREAD

**Discussion by lipeng28** [Ericsson/codechecker#2841] [Q&A] [answered]
Hi There

When I applied the command: CodeChecker server -w ./ws -v 8555, I met the following error message:

Couldn't import module for subcommand 'server'... ignoring.
Traceback (most recent call last):
  File "/Users/pengli/CodeChecker-byd/build/CodeChecker/cc_bin/CodeChecker.py", line 110, in main
    subcommands[subcommand])
  File "/Users/pengli/CodeChecker-byd/build/CodeChecker/cc_bin/CodeChecker.py", line 39, in add_subcommand
    command_module = cmd_spec.loader.load_module(module_name)
  File "<frozen importlib._bootstrap_external>", line 407, in _check_name_wrapper
  File "<frozen importlib._bootstrap_external>", line 907, in load_module
  File "<frozen importlib._bootstrap_external>", line 732, in load_module
  File "<frozen importlib._bootstrap>", line 265, in _load_module_shim
  File "<frozen importlib._bootstrap>", line 696, in _load
  File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 728, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/Users/pengli/CodeChecker-byd/build/CodeChecker/cc_bin/../lib/python3/codechecker_server/cmd/server.py", line 27, in <module>
    from codechecker_api_shared.ttypes import DBStatus
ModuleNotFoundError: No module named 'codechecker_api_shared'
usage: CodeChecker [-h] {} ...
CodeChecker: error: invalid choice: 'server' (choose from )

Any idea about it? 

Thanks
Peng

**Comment by lipeng28** [comment]
I searched the entire code base, and did not find "codechecker_api_shared" module, can you please advise if it is missed?

Thanks
Peng

**Comment by csordasmarton** [comment]
@lipeng28 There was a huge code refactoring in the v6.12.0 release where we moved the generated thrift API stubs to python/nodejs modules. See: #2570. So `codechecker_api_shared` module can be found on the pypi (https://pypi.org/project/codechecker-api-shared/) and it is listed in the python requirements files (https://github.com/Ericsson/codechecker/search?l=Text&q=codechecker_api_shared) and it will be installed to your virtual environment on `make venv` or `make venv_dev` targets. I think your problem is that you are using an old virtual environment where this module is not installed. Could you please try to run the following commands:

```sh
# Clear your directory.
make clean

# Create a Python virtualenv and set it as your environment.
make venv  # This command will install the codechecker_api_shared module too.
source $PWD/venv/bin/activate

# Build and install a CodeChecker package.
make package

```

**Comment by lipeng28** [comment]
@csordasmarton I tried your instructions, however, it does not work. I still met the following error:
ModuleNotFoundError: No module named 'codechecker_api_shared'

Thanks
Peng

**Comment by csordasmarton** [comment]
@lipeng28 Could you please provide us a few more information:
- Your OS type and version.
- Please run the following commands inside your codechecker directory and give us the outputs:
  - `git rev-parse HEAD`
  - `rm -rf venv && make venv`
  - `pip list | grep codechecker`

**Comment by lipeng28** [comment]
@csordasmarton 

OS type and version: MacOS version 10.15.5

git rev-parse HEAD: e59c1b47dd821aed702bf8701d07264fe25b9985
rm -rf venv && make venv
pip list | grep codechecker
codechecker-api        6.27.0
codechecker-api-shared 6.27.0

Looks like codechecker-api-shared is installed, however, CodeChecker server does not work fine. 

Best,
Peng

**Comment by gyorb** [comment, marked as answer]
Hi,

Is is possible that the virtualenv is not sourced and the codechecker-api modules are installed on the system level too (v6.27 is an older version)?
```
rm -rf venv && make venv
source venv/bin/activate
pip list | grep codechecker
```

I did not found this commit on the master branch:
```
git show e59c1b47dd821aed702bf8701d07264fe25b9985
fatal: bad object e59c1b47dd821aed702bf8701d07264fe25b9985
```

Best,
Gyorgy

**Comment by lipeng28** [comment]
@csordasmarton 
it works now.
As for the git head, It is because I had my own repo to support my own extension.

Thanks
Peng
